### PR TITLE
WinForms support in unit tests when targeting .Net Core

### LINKED ISSUE #3426: Add WinForms support in unit tests when targeting .Net Core
Some tests, like `DoNotCallExitMethodsTest` reference `WinForms` types and currently cannot be run while targeting .Net Core.

### PR DESCRIPTION
**Pull request by csaba-sagi-sonarsource**
Fixes #3426

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/Helpers/RemovableDeclarationCollectorTest.cs [hunk 1]
```diff
@@ -20,7 +20,6 @@
 
 using System.Linq;
 using FluentAssertions;
-using Microsoft.CodeAnalysis;
```
Comment (andrei-epure-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Helpers/RemovableDeclarationCollectorTest.cs:23]: why this change?

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MarkWindowsFormsMainWithStaThread.vb [hunk 2]
```diff
@@ -14,8 +14,6 @@ Namespace Tests.TestCases
   Public Class Program_01
     Shared Sub Main() ' Noncompliant {{Add the 'STAThread' attribute to this entry point.}}
 '              ^^^^
-      Dim winForm As Form = New Form
```
Comment (andrei-epure-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MarkWindowsFormsMainWithStaThread.vb:17]: why remove this?

**Review by andrei-epure-sonarsource — CHANGES_REQUESTED** [review]
I don't understand some changes, could you please explain?

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MarkWindowsFormsMainWithStaThread.vb (hunk 2 above)
Comment (andrei-epure-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MarkWindowsFormsMainWithStaThread.vb:17]: or is it just looking at the namespace?

Comment (csaba-sagi-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MarkWindowsFormsMainWithStaThread.vb:17]: After some offline talk with Andrei, we decided there is no action to be taken.

**Review by andrei-epure-sonarsource — APPROVED** [review]
LGTM
